components/me.js

```diff
@@ -8,15 +8,25 @@ export const MeContext = React.createContext({
 })
 
 export function MeProvider ({ me, children }) {
-  const { data } = useQuery(ME, SSR ? {} : { pollInterval: 1000, nextFetchPolicy: 'cache-and-network' })
+  const { data, refetch } = useQuery(ME, SSR ? {} : { pollInterval: 1000, nextFetchPolicy: 'cache-and-network' })
+  // this makes sure that we always use the fetched data if it's null.
+  // without this, we would always fallback to the `me` object
+  // which was passed during page load which (visually) breaks switching to anon
+  const futureMe = data?.me ?? (data?.me === null ? null : me)
 
   return (
-    <MeContext.Provider value={data?.me || me}>
+    <MeContext.Provider value={{ me: futureMe, refetch }}>
       {children}
     </MeContext.Provider>
   )
 }
 
 export function useMe () {
-  return useContext(MeContext)
+  const { me } = useContext(MeContext)
+  return me
+}
+
+export function useMeRefresh () {
+  const { refetch } = useContext(MeContext)
+  return refetch
```

Comment:
Just a nit but it might be better to either refactor `useMe` to return the full context object *or* have a `useMeFull` that returns the full context object so that we can avoid creating more getters for the context.

Reply:
Makes sense, will do!

Reply:
Decided to go with the backwards incompatible change since I think a simple search and replace is enough to prevent bugs: bf7ec18